Panic when purchasing a cryptocurrency listing

The following example produced a panic when attempting to be purchased.
```
10:02:33.821 [ERROR] [core/CalculateOrderTotal] Currency not tracked
goroutine 61423 [running]:
runtime/debug.Stack(0x49e84eb, 0xc000336ed0, 0x1)
    /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
    /usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/OpenBazaar/openbazaar-go/api.(*jsonAPIHandler).ServeHTTP.func1()
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/api/jsonapi.go:177 +0x10b
panic(0x53486e0, 0x697eee0)
    /usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/OpenBazaar/openbazaar-go/core.(*OpenBazaarNode).CalculateOrderTotal(0xc001213200, 0xc0001218c0, 0x4, 0x5a493e0, 0xc001267420)
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/core/order.go:967 +0xcd6
github.com/OpenBazaar/openbazaar-go/core.(*OpenBazaarNode).Purchase(0xc001213200, 0xc000121800, 0xc000121800, 0x0, 0x0, 0x210, 0x0, 0x0, 0xc0005bbb68, 0xc00211f948)
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/core/order.go:160 +0x6ae
github.com/OpenBazaar/openbazaar-go/api.(*jsonAPIHandler).POSTPurchase(0xc001312900, 0x5a2d180, 0xc001451dc0, 0xc000626c00)
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/api/jsonapi.go:595 +0x125
github.com/OpenBazaar/openbazaar-go/api.(*jsonAPIHandler).POSTPurchase-fm(0x5a2d180, 0xc001451dc0, 0xc000626c00)
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/api/endpoints.go:101 +0x48
github.com/OpenBazaar/openbazaar-go/api.blockingStartupMiddleware(0xc001312900, 0x5a2d180, 0xc001451dc0, 0xc000626c00, 0xc00211f9f8)
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/api/endpoints.go:270 +0x62
github.com/OpenBazaar/openbazaar-go/api.post(0xc001312900, 0xc0016596f0, 0xc, 0x5a2d180, 0xc001451dc0, 0xc000626c00)
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/api/endpoints.go:101 +0x7fe
github.com/OpenBazaar/openbazaar-go/api.(*jsonAPIHandler).ServeHTTP(0xc001312900, 0x5a2d180, 0xc001451dc0, 0xc000626c00)
    /Users/rob/develop/openbazaar/v2/go/src/github.com/OpenBazaar/openbazaar-go/api/jsonapi.go:186 +0x8a3
net/http.(*ServeMux).ServeHTTP(0xc0012d9440, 0x5a2d180, 0xc001451dc0, 0xc000626c00)
    /usr/local/go/src/net/http/server.go:2361 +0x127
net/http.serverHandler.ServeHTTP(0xc0012ff450, 0x5a2d180, 0xc001451dc0, 0xc000626c00)
    /usr/local/go/src/net/http/server.go:2741 +0xab
net/http.(*conn).serve(0xc002ea8320, 0x5a2ed40, 0xc00124eec0)
    /usr/local/go/src/net/http/server.go:1847 +0x646
created by net/http.(*Server).Serve (edited) 
/usr/local/go/src/net/http/server.go:2851 +0x2f5
```

The listing:
```
{
    "listing": {
        "slug": "btc-mhc",
        "vendorID": {
            "peerID": "Qmdp9Q2UzitxeUwc5EbAKAEc7Qhodm4C5c8m13dsHMZDQw",
            "handle": "",
            "pubkeys": {
                "identity": "CAESIBBPAJGzlaA//J1pZAvPKEYqH0gClpruznP9wLol+lBR",
                "bitcoin": "AkN8Z+yv1fh6b6klk/2HYlcy6e09ziNqOySn/9Duwt74"
            },
            "bitcoinSig": "MEQCIGya0oP/rkaSz36/IqIEK8eSaSAQ1zeX1hTI45mbJbGDAiAAp6zZ0bye4tJLTVRD6BRl4kYxyTzu1UWVihXwP7vAzA=="
        },
        "metadata": {
            "version": 5,
            "contractType": "CRYPTOCURRENCY",
            "format": "MARKET_PRICE",
            "expiry": "2037-12-31T06:00:00.000Z",
            "acceptedCurrencies": [
                "TBTC"
            ],
            "pricingCurrency": "",
            "language": "",
            "escrowTimeoutHours": 0,
            "coinType": "MHC",
            "coinDivisibility": 8,
            "priceModifier": 0
        },
        "item": {
            "title": "BTC-MHC",
            "description": "",
            "processingTime": "",
            "price": 0,
            "nsfw": false,
            "tags": [
            ],
            "images": [
                {
                    "filename": "alvinshow01.gif",
                    "original": "Qmc6qbG2jV5dtfSJhzHCxgJJGjABCHQmXGJcDKuGmswV4p",
                    "large": "QmYKDiFCMYR9pirv99zGm6kHZ2s5LRq55j31BLbbuSKxqA",
                    "medium": "QmUYd1VrE9tcGe67ZW8o3E2DEiJ8PurTE45yiVucfhaSmG",
                    "small": "QmcPuwmTMRz6im32HXYTcaXPpzPb4pzVTDBycr8USBXQ6y",
                    "tiny": "QmaJKHgeidwWQ2kzy4VwE8cvGwo64QYLSA5mEosiu9ZmoT"
                }
            ],
            "categories": [
            ],
            "grams": 0,
            "condition": "",
            "options": [
            ],
            "skus": [
                {
                    "productID": "",
                    "surcharge": 0,
                    "quantity": 0,
                    "bigSurcharge": "",
                    "bigQuantity": "10000000000"
                }
            ],
            "priceModifier": 3,
            "bigPrice": ""
        },
        "shippingOptions": [
        ],
        "coupons": [
        ],
        "moderators": [
        ],
        "termsAndConditions": "",
        "refundPolicy": ""
    },
    "hash": "QmRQqDD2awvHsZi37NVmTvGB6SDWiveuDGM3z6Xg3UqGTH",
    "signature": "xHhXs1bYnMp2DVIQ03fhc1oVZ9zISjKBFC07X7yHLjolBwnZXW1U46/VG/yEYSWrDhNAaavH9aVkVBNP6NVnBA=="
}
```

The purchase request:
```
{
  "address": "",
  "addressNotes": "",
  "alternateContactInfo": "",
  "city": "",
  "countryCode": "",
  "items": [
    {
      "listingHash": "QmRQqDD2awvHsZi37NVmTvGB6SDWiveuDGM3z6Xg3UqGTH",
      "options": [],
      "shipping": {
        "name": "",
      },
      "memo": "",
      "coupons": [],
      "bigQuantity": "1000000",
      "paymentAddress": "howdy-addr",
    }
  ],
  "moderator": "",
  "paymentCoin": "TBTC",
  "postalCode": "",
  "shipTo": "",
  "state": "",
}
```

--- a/core/order.go
+++ b/core/order.go
@@ -964,7 +964,7 @@ func (n *OpenBazaarNode) CalculateOrderTotal(contract *pb.RicardianContract) (*b
 			} else if l.Item.PriceModifier != 0 {
 				priceModifier = float64(l.Item.PriceModifier)
 			}
-			satoshis, err = n.getMarketPriceInSatoshis(contract.BuyerOrder.Payment.AmountCurrency.Code, l.Item.PriceCurrency.Code, GetOrderQuantity(l, item))
+			satoshis, err = n.getMarketPriceInSatoshis(contract.BuyerOrder.Payment.AmountCurrency.Code, l.Metadata.CryptoCurrencyCode, GetOrderQuantity(l, item))
 			if err != nil {
 				return big.NewInt(0), err
 			}

--- a/core/order_test.go
+++ b/core/order_test.go
@@ -22,7 +22,7 @@ func TestOpenBazaarNode_CalculateOrderTotal(t *testing.T) {
 				ContractType:       pb.Listing_Metadata_PHYSICAL_GOOD,
 				Format:             pb.Listing_Metadata_FIXED_PRICE,
 				AcceptedCurrencies: []string{"TBTC"},
-				Version:            2,
+				Version:            5,
 			},
 			Item: &pb.Listing_Item{
 				BigPrice:      "100000",
@@ -57,7 +57,7 @@ func TestOpenBazaarNode_CalculateOrderTotal(t *testing.T) {
 		Items: []*pb.Order_Item{
 			{
 				ListingHash: listingID.String(),
-				Quantity:    1,
+				BigQuantity: "1",
 				ShippingOption: &pb.Order_Item_ShippingOption{
 					Name:    "UPS",
 					Service: "Standard shipping",
@@ -83,7 +83,7 @@ func TestOpenBazaarNode_CalculateOrderTotal(t *testing.T) {
 	}
 
 	// Test higher quantity
-	contract.BuyerOrder.Items[0].Quantity = 2
+	contract.BuyerOrder.Items[0].BigQuantity = "2"
 	total, err = node.CalculateOrderTotal(contract)
 	if err != nil {
 		t.Error(err)
@@ -93,7 +93,7 @@ func TestOpenBazaarNode_CalculateOrderTotal(t *testing.T) {
 	}
 
 	// Test with options
-	contract.BuyerOrder.Items[0].Quantity = 1
+	contract.BuyerOrder.Items[0].BigQuantity = "1"
 	contract.VendorListings[0].Item.Options = []*pb.Listing_Item_Option{
 		{
 			Name: "color",

